core/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java

```diff
@@ -407,4 +408,33 @@ public void testIndexMetaField() throws IOException {
             assertThat(query, instanceOf(MatchAllDocsQuery.class));
         }
     }
+
+    public void testToQueryPrefixQuery() throws Exception {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
+        for (Operator op : Operator.values()) {
+            Query query = new SimpleQueryStringBuilder("foo-bar-foobar*")
+                .field(STRING_FIELD_NAME)
+                .analyzeWildcard(true)
+                .analyzer("standard")
+                .defaultOperator(op)
+                .toQuery(createShardContext());
+            assertThat(query, instanceOf(BooleanQuery.class));
+            BooleanQuery bq = (BooleanQuery) query;
+            assertThat(bq.clauses().size(), CoreMatchers.equalTo(3));
+            String[] expectedTerms = new String[]{"foo", "bar", "foobar"};
+            for (int i = 0; i < bq.clauses().size(); i++) {
+                BooleanClause clause = bq.clauses().get(i);
+                if (i != bq.clauses().size() - 1) {
+                    assertTermQuery(clause.getQuery(), STRING_FIELD_NAME, expectedTerms[i]);
+                } else {
+                    assertPrefixQuery(clause.getQuery(), STRING_FIELD_NAME, expectedTerms[i]);
+                }
+                if (op == Operator.AND) {
+                    assertThat(clause.getOccur(), CoreMatchers.equalTo(BooleanClause.Occur.MUST));
+                } else {
+                    assertThat(clause.getOccur(), CoreMatchers.equalTo(BooleanClause.Occur.SHOULD));
+                }
+            }
+        }
+    }
```

Comment:
That depends a bit on how much it complicates the setup of AbstractQueryTestCase. I find that setup a bit too large and complex already. If there is an easy way to add a custom analyzer to the current setup I think it would be okay.